modules/openapi-generator/src/main/resources/kotlin-client/enum_class.mustache

```diff
@@ -52,12 +52,29 @@ import kotlinx.serialization.internal.CommonEnumSerializer
     {{&name}}({{{value}}}){{^-last}},{{/-last}}{{#-last}};{{/-last}}
     {{/isArray}}
 {{/enumVars}}{{/allowableValues}}
-
     /**
-    This override toString avoids using the enum var name and uses the actual api value instead.
-    In cases the var name and value are different, the client would send incorrect enums to the server.
-    **/
-    override fun toString(): String {
-        return value{{^isString}}.toString(){{/isString}}
+     * Override toString() to avoid using the enum variable name as the value, and instead use
+     * the actual value defined in the API spec file.
+     *
+     * This solves a problem when the variable name and its value are different, and ensures that
+     * the client sends the correct enum values to the server always.
+     */
+    override fun toString(): String = value{{^isString}}.toString(){{/isString}}
+
+    companion object {
+        /**
+         * Converts the provided [data] to a [String] on success, null otherwise.
+         */
+        fun encode(data: Any?): kotlin.String? = if (data is {{classname}}) "$data" else null
```

Comment:
@noordawod could you please explain what is the use case for this utility?

Reply:
Unused by me, but because `decode()` is introduced I wanted to compliment it with an `encode()`, I think it's a good practice.

Reply:
ps: I use `decode()`.